config/software/ruby-windows.rb

```diff
@@ -55,16 +49,15 @@
   # to turn off the warning message
   # We are only removing dl.rb:8
   # => warn "DL is deprecated, please use Fiddle"
-  if version.start_with? "2"
-    block do
-      require 'digest/md5'
+  block do
+    require 'digest/md5'
 
-      ABI_ver = version[/(^\d+\.\d+)/] + '.0'
-      dl_path = File.join(install_dir, "embedded/lib/ruby", ABI_ver, "dl.rb")
+    ABI_ver = version[/(^\d+\.\d+)/] + '.0'
+    dl_path = File.join(install_dir, "embedded/lib/ruby", ABI_ver, "dl.rb")
 
-      if Digest::MD5.hexdigest(File.read(dl_path)) == "78c185a3fcc7b5e2c3db697c85110d8f"
-        File.open(dl_path, "w") do |f|
-          f.print <<-E
+    if Digest::MD5.hexdigest(File.read(dl_path)) == "78c185a3fcc7b5e2c3db697c85110d8f"
```

Comment:
There is a digester class in Omnibus and we should use that here if possible. Reading an entire downloaded file into memory just to md5 it is really inefficient :smile: